google-cloud-spanner/test/google/cloud/spanner/client/read_resume_buffer_bound_test.rb

```diff
@@ -184,7 +184,7 @@
       Google::Cloud::Spanner::V1::PartialResultSet.new(results_hash1),
       Google::Cloud::Spanner::V1::PartialResultSet.new(results_hash2),
       Google::Cloud::Spanner::V1::PartialResultSet.new(results_hash3),
-      GRPC::Unavailable,
+      Google::Cloud::AbortedError,
```

Comment:
Why does the old code work then? Where does it convert to `Google::Cloud::UnavailableError`?

Reply:
The conversion happens here: https://github.com/googleapis/google-cloud-ruby/blob/e4bcb6ec8b8b525363b4add6f45516256aa33489/google-cloud-spanner/lib/google/cloud/spanner/results.rb#L149-L153

Reply:
In your changes (google-cloud-spanner/lib/google/cloud/spanner/results.rb), Line 176 will do the same conversion.

Reply:
But you're right. We can assume that all `GRPC::BadStatus` are converted to `Google::Cloud::Error` in the gapic layer.